From 4fdbe05281bfc3c9e151b6035f17651679354971 Mon Sep 17 00:00:00 2001
From: Izak Burger <iburger@victronenergy.com>
Date: Fri, 1 Dec 2023 09:39:17 +0200
Subject: [PATCH] comap: Fix engine runtime register scale factor

Engine runtime for ComAp is from COM.Obj 8206 which is in hours
with one decimal. Therefore multiply by 6 to get minutes, and 60
to get seconds, or scale=1/360.

---
 comap.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/comap.py b/comap.py
index a7b88f8..797aa79 100644
--- a/comap.py
+++ b/comap.py
@@ -53,7 +53,7 @@ class ComAp_Generator(device.ModbusDevice):
             Reg_s16(1006, '/Engine/CoolantTemperature', 10, '%.1f C', invalid=-0x8000),
             Reg_s16(1008, '/Engine/OilPressure',         1, '%.0f kPa', invalid=-0x8000),
             Reg_u16(1010, '/Engine/Load',                1, '%.0f %%', invalid=0x8000),
-            Reg_u32b(1013, '/Engine/OperatingHours',   1/6, '%.1f s', invalid=0x80000000),
+            Reg_u32b(1013, '/Engine/OperatingHours', 1/360, '%.1f s', invalid=0x80000000),
             Reg_u16(1053, '/StarterVoltage',            10, '%.1f V'),
             Reg_u16(1055, '/FuelLevel',                  1, '%.0f %%', invalid=0x8000),
 
